### DISCUSSION TITLE
FB4D over Proxy

### DISCUSSION THREAD

**Discussion by zarkogajic** [SchneiderInfosystems/FB4D#60] [Ideas]
Hi,

How does FB4D handle connections over a Proxy server?

p.s.
Very new to Firebase and exploring possible options to have Firebase / FB4D in my VCL project.

-žarko

**Comment by SchneiderInfosystems** [comment]
Hi Zarko

FB4D uses internally THTTPClient for synchronous access and within the TRealTimeDB.ListenForValueEvents and TRESTClient for asynchronous access. 

For TRESTClient proxy support is implemented with settings for Server, Post, UserName, and Password offers. So I could easily integrate these four properties into the TFirebaseConfiguration and assign these values each time I create a new RESTClient.

For  THTTPClient I could use THTTPClient.ProxSettings to do the same. Here I see more properties for Scheme and Credentials. Do we need that too?

I'm currently very busy and this feature is the first time requested so I'm unsure when this feature could be available.
If you are able to integrate that by yourself in a forked repository, please send me a pull request so I can check to overtake the changes. I would prefer a new method in IFirebaseConfiguration like this:

procedure SetupProxy(Server: string; Port: integer; User, Password: string);

best regards,
Christoph

**Comment by zarkogajic** [comment]
Thanks Chris,

Ok, thanks. I'll first have to make sure Firebase is the way to go for my project.
If so, and if Proxy would be a must, I'll start playing around - and we'll see how that goes...

-žarko

**Reply by ChristophSchneiderTrihow** [reply]
Please read also this: Chapter **_Handling System Proxy Settings With an HTTP Client_** in
http://docwiki.embarcadero.com/RADStudio/Sydney/en/Using_an_HTTP_Client. 
In Short; The HTTP-Client uses the system proxy settings: So you can use the default proxy from the underlying system. We would need only an own solution if you want to configure the proxy application-specific.

**Reply by SchneiderInfosystems** [reply]
I believe we can close this discussion after my explanations, right? Or is there still a demand for proxy configuration?

**Reply by zarkogajic** [reply]
Yes, we can close. I've implemented the stuff I need differently...

-žarko
